src/Transports.AspNetCore/WebSockets/BaseSubscriptionServer.cs

```diff
@@ -0,0 +1,488 @@
+using System.Security.Claims;
+using System.Security.Principal;
+using GraphQL.Execution;
+using GraphQL.Transport;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// Manages a WebSocket message stream.
+/// </summary>
+public abstract partial class BaseSubscriptionServer : IOperationMessageProcessor
+{
+    private volatile int _initialized;
+    private CancellationTokenSource? _cancellationTokenSource;
+    private readonly GraphQLHttpMiddlewareOptions _options;
+
+    /// <summary>
+    /// Returns a <see cref="IWebSocketConnection"/> instance that can be used
+    /// to send messages to the client.
+    /// </summary>
+    protected IWebSocketConnection Client { get; }
```

Comment:
```suggestion
    protected IWebSocketConnection Connection { get; }
```

I would rename. It looks more familiar.